test/test.cc

```diff
@@ -388,18 +381,18 @@ boost::property_tree::ptree make_config(const std::string& path_prefix,
   // force the paths to be different
   boost::replace_all(defaults, "%%", path_prefix);
 
-  // make ptree and override defaults
```

Comment:
yeah config singleton is a double edged sword. on one hand using makes life way easier for 99.99% of all use-cases. but then in testing, if we need to change the configs to test something and the singleton is used throughout the library it will be a gotcha. im all for crossing that bridge when we get there (i have an idea for how to make it work)

Reply:
for tests I think it's fine though right? bcs we can change things with `overrides`  & `removes` when we call `make_config`.

Reply:
once there are calls to the config singleton embedded deep in the library code, a test wont be able to control which which one it will get. it will always get the first config it was called with. so in the future when that is a problem we'll need a way to inject a change overtop of the singletons initial copy